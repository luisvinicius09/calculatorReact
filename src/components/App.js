import React, { useState } from 'react';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import Calculate from '../logic/calculate';

const App = () => {
  const [results, setResults] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    const calculation = Calculate(results, buttonName);
    setResults(calculation);
  };

  const { total, next, operation } = results;
  let value = total;
  if (operation) {
    value = operation;
    if (next) {
      value = next;
    }
  }

  const styles = {
    display: 'flex',
    width: '700px',
    flexDirection: 'column',
    margin: 'auto',
    position: 'relative',
    top: '100px',
  };

  return (
    <>
      <div style={styles}>
        <Display result={value || '0'} />
        <ButtonPanel clickHandler={handleClick} />
      </div>
    </>
  );
};

export default App;
