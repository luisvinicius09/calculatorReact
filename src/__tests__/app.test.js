import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../components/App';

it('renders the app correctly', () => {
  const app = renderer.create(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  ).toJSON();
  expect(app).toMatchSnapshot();
});

it('should render the number 1', () => {
  const { getByText } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const btnName = getByText(/1/);
  expect(btnName).toBeInTheDocument();
});

it('should render the number 2', () => {
  const { getByText } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const btnName = getByText(/2/);
  expect(btnName).toBeInTheDocument();
});

it('should render the number 3', () => {
  const { getByText } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const btnName = getByText(/3/);
  expect(btnName).toBeInTheDocument();
});

it('should render the number 4', () => {
  const { getByText } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const btnName = getByText(/4/);
  expect(btnName).toBeInTheDocument();
});

it('should render the number 5', () => {
  const { getByText } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const btnName = getByText(/5/);
  expect(btnName).toBeInTheDocument();
});

it('should render the number 6', () => {
  const { getByText } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const btnName = getByText(/6/);
  expect(btnName).toBeInTheDocument();
});

it('should render the number 7', () => {
  const { getByText } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const btnName = getByText(/7/);
  expect(btnName).toBeInTheDocument();
});

it('should render the number 8', () => {
  const { getByText } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const btnName = getByText(/8/);
  expect(btnName).toBeInTheDocument();
});

it('should render the number 9', () => {
  const { getByText } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const btnName = getByText(/9/);
  expect(btnName).toBeInTheDocument();
});
