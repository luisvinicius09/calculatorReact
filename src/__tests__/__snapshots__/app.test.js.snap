// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app correctly 1`] = `
Array [
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "marginLeft": "10px",
        }
      }
    >
      <h1>
        Math Magicians
      </h1>
    </div>
    <div
      style={
        Object {
          "margin": "30px",
        }
      }
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "paddingRight": "10px",
          }
        }
      >
        Home
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
        style={
          Object {
            "paddingRight": "10px",
          }
        }
      >
        Calculator
      </a>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote of the day
      </a>
    </div>
  </div>,
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "margin": "auto",
        "position": "relative",
        "top": "100px",
        "width": "700px",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "grey",
          "color": "white",
          "display": "flex",
          "flexDirection": "column",
          "height": "100px",
          "padding": "0 10px 0 10px",
          "textAlign": "right",
        }
      }
    >
      <h1>
        0
      </h1>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexWrap": "wrap",
          "height": "400px",
        }
      }
    >
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#fb8500",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        A/C
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#fb8500",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#fb8500",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#fb8500",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        รท
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#fb8500",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#fb8500",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        +
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#fb8500",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 50%",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#edf6f9",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#fb8500",
            "flex": "1 1 25%",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
